plugins/texteditor.koplugin/main.lua

```diff
@@ -0,0 +1,160 @@
+local DataStorage = require("datastorage")
+local InfoMessage = require("ui/widget/infomessage")
+local InputDialog = require("ui/widget/inputdialog")
+local PathChooser = require("ui/widget/pathchooser")
+local UIManager = require("ui/uimanager")
+local WidgetContainer = require("ui/widget/container/widgetcontainer")
+local logger = require("logger")
+local util = require("ffi/util")
+local _ = require("gettext")
+local Screen = require("device").screen
+
+
+local TextEditor = WidgetContainer:new{
+    name = "text_editor",
+}
+
+function TextEditor:init()
+    self.ui.menu:registerToMainMenu(self)
+end
+
+function TextEditor:start(file_path)
+    local FileManager = require("apps/filemanager/filemanager")
+    FileManager:onClose()
+    if not file_path then
+        self.context = ""
+        self.file_path = ""
+    else
+        self.file_path = file_path
+        self:readFile(file_path)
+    end
+    self.input = InputDialog:new{
+        title =  _("Basic text editor"),
+        input = self.context,
+        text_height = Screen:getHeight() * 0.5,
+        width = self.width or Screen:getWidth(),
+        input_type = "string",
+        enter_callback = function ()
+            return '\n'
+        end,
+        buttons = {{
+        {
+            text = _("Open"),
+            callback = function()
+                self:chooseFile()
+            end,
+        },
+        {
+            text = _("Save"),
+            callback = function()
+                self:saveFile(self.file_path)
+            end,
+        },{
+            text = _("Quit"),
+            callback = function()
+                UIManager:show(ConfirmBox:new{
+                    text = _("Are you sure that you want to quit editor? All unsaved changes will be lost."),
+                    ok_text = _("Quit"),
+                    ok_callback = function()
+                        self.context = ""
+                        self.file_path = ""
+                        UIManager:close(self.input)
+                    end
+                })
+
+            end,
+        },
+        {
+            text = _("Pg_up"),
+            callback = function()
+                self.input._input_widget.text_widget:scrollUp()
+                UIManager:setDirty(self.input, function()
+                    return "ui", self.input.dimen
+                end)
+            end,
+        },
+        {
+            text = _("Pg_dwn"),
+            callback = function()
+                self.input._input_widget.text_widget:scrollDown()
+                UIManager:setDirty(self.input, function()
+                    return "ui", self.input.dimen
+                end)
+            end,
+        }
+
+    }},
+    }
+    UIManager:show(self.input)
+    self.input:onShowKeyboard()
+end
+
+
+function TextEditor:chooseFile()
+    self.input:onClose()
+    local path_chooser = PathChooser:new{
+        title = _("Choose file. Long press to confirm"),
+        height = Screen:getHeight(),
```

Comment:
Does that leave room for padding and margin?

Reply:
![Uploading Zrzut ekranu z 2018-07-11 13-11-37.png…]()